frontend/packages/dev-console/src/components/NamespacedPage.tsx

```diff
@@ -28,23 +29,28 @@ const NamespacedPage: React.FC<NamespacedPageProps> = ({
   variant = NamespacedPageVariants.default,
   toolbar,
 }) => (
-  <div className="odc-namespaced-page">
-    <NamespaceBar
-      disabled={disabled}
-      onNamespaceChange={onNamespaceChange}
-      hideProjects={hideProjects}
-    >
-      {!hideApplications && <NamespaceBarApplicationSelector disabled={disabled} />}
-      {toolbar && <div className="odc-namespaced-page__toolbar">{toolbar}</div>}
-    </NamespaceBar>
-    <div
-      className={cx('odc-namespaced-page__content', {
-        [`is-${variant}`]: variant !== NamespacedPageVariants.default,
-      })}
-    >
-      {children}
-    </div>
-  </div>
+  <FormScrollShadowProvider>
```

Comment:
I don't think we need to wrap the whole namespaced page component into this scroll provider. This component will soon move out to either `console-shared` or `sdk` package. Also the name `FormScrollShadowProvider` kind of hints that this component is going to be used for just forms.

Reply:
I think we can move it to the ImportPage component which deals with all the forms. 
We can rename it to `ScrollShadowProvider`. WDYT?

Reply:
By wrapping every namespace page, we're going to get a bunch of react updates every time we are scrolling pages that don't even care about the scroll position.

Ideally I'd like a to let each component manage their own needs. Which means that the footer should identify the scroll container and attach its own listener.

There's a similar need which was addressed in `dom-utils` to find the parent scroll container. Perhaps we can apply the same concept to a new hook and let the footer `find` the scroll parent to listen to. The hook can return `[scrollContainerRef, setElementRef]` where `setElementRef` is applied to the footer and the hook will traverse up the dom to find the scroll container.

Reply:
used approach mentioned by @christianvogt which works fine.